restli-common/src/main/java/com/linkedin/restli/internal/common/URIParamUtils.java

```diff
@@ -62,7 +70,14 @@
     {
       if (dataMap.containsKey(parameterName))
       {
-        result.put(parameterName, URIMaskUtil.encodeMaskForURI(dataMap.getDataMap(parameterName)));
+        if (dataMap.get(parameterName) instanceof String)
```

Comment:
Minor comments:
1. Add a final `else` case which throws an exception. If one of these projection params is present but is not a string or data map, then it should be treated as an illegal argument.
2. A minor optimization could be made here. You have 4 different `get` accessor calls here, you could do one initial get then perform the checks/casting on that object.

Reply:
Fixed